### PR TITLE
Gateway

### PR DESCRIPTION
**Pull request by auperman-lab**
finnaly gateway

### CODE REVIEW

**Comment by catness812** [comment]
why did you make changes in petition_service? specifically the db port

**Comment by auperman-lab** [comment]
sorry , nu se primea de unit la petitions si asa o schimbat, eu nu am observat si am lasat asa

**Comment by auperman-lab** [comment]
restored it(sorry)

**Comment by catness812** [comment]
it's okay, presupun ca nu se primea de conectat la petitions pt ca am vazut voi ati schimbat grpc client ports de conectare, respectiv eu la fel trb sa schimb grpc server port la petitions

File: gateway/internal/petition/petition_controller.go [hunk 1]
```diff
@@ -28,32 +26,66 @@ func NewPetitionController(service IPetitionService) IPetitionController {
...
-func (c *petitionController) GetPetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
+	ctx.JSON(http.StatusOK, msg)
 
 }
 
-func (c *petitionController) DeletePetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
-}
+func (c *petitionController) GetPetitions(ctx *gin.Context) {
```
Comment (madalv) [inline gateway/internal/petition/petition_controller.go:63]: Pagination parameters have to be taken from the query params, not from body

File: gateway/internal/petition/petition_repository.go [hunk 2]
```diff
@@ -27,17 +30,83 @@ type petitionRepository struct {
...
+	}
+
+	message = resp.Message
+
+	return message, nil
 
 }
 
-func (svc *petitionRepository) Update() {
+func (repo *petitionRepository) DeletePetition(id uint32) (string, error) {
```
Comment (madalv) [inline gateway/internal/petition/petition_repository.go:99]: Why not just get the petition id from the URL? Why are you binding a whole petition from the request body?

File: gateway/internal/petition/petition_repository.go [hunk 3]
```diff
@@ -27,17 +30,83 @@ type petitionRepository struct {
...
+		},
+	})
+
+	if err != nil {
+		return 0, err
+	}
+	return resp.PetitionId, nil
+}
+
+func (repo *petitionRepository) GetPetitions(query model.PaginationQuery) ([]model.Petition, error) {
```
Comment (madalv) [inline gateway/internal/petition/petition_repository.go:64]: Why not just get the petition id from the URL? Why are you binding a whole petition from the request body?

File: petition_service/internal/config/config.go [hunk 4]
```diff
@@ -23,7 +23,7 @@ type Postgres struct {
 var Cfg Config
 
 func LoadConfig() {
-	data, err := os.ReadFile("petition_service/config.yml")
+	data, err := os.ReadFile("./config.yml")
```
Comment (catness812) [inline petition_service/internal/config/config.go]: not sure what was the intention, dar aici la fel nu trebuie de schimbat nimic; LoadConfig deja e ajustat pentru petitii

File: gateway/internal/petition/petition_controller.go [hunk 5]
```diff
@@ -28,32 +26,66 @@ func NewPetitionController(service IPetitionService) IPetitionController {
...
+
+	ctx.JSON(http.StatusOK, petitions)
 
-func (c *petitionController) SignPetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
 }
 
-func (c *petitionController) SetStatus(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
+func (c *petitionController) DeletePetition(ctx *gin.Context) {
```
Comment (madalv) [inline gateway/internal/petition/petition_controller.go:81]: Why not just get the petition id from the URL? Why are you binding a whole petition from the request body?

Comment (Masha003) [reply gateway/internal/petition/petition_controller.go:81]: I know, m-am condus după proto messages când am făcut, dar poate nu am înțeles corect. Voi corecta când vin la oficiu

**Review by ktruedat — APPROVED** [review]

**Review by grumpycatyo-collab — APPROVED** [review]

**Review by valeis — APPROVED** [review]

**Pull request merged**